src/sections/Learn-Layer5/Chapters/index.js

```diff
@@ -7,62 +7,82 @@ import Image from "../../../components/image";
 import { ChapterWrapper } from "./chapters.style";
 import ReactTooltip from "react-tooltip";
 
-const Chapters = ({chapterData, courseData, location, serviceMeshesList}) => {
+const Chapters = ({chapterData, courseData, location, serviceMeshesList, TOCData}) => {
+
   const { frontmatter, body } = chapterData;
   const serviceMeshImages = courseData.frontmatter.meshesYouLearn;
-  
+  const tableOfContents = TOCData
+    .filter(node => !!node.fields.section)
+    .map( toc => ({section: toc.fields.section, chapter: toc.fields.chapter}) );
+
+
   const replaceSlugPart = (index) => (oldSlug) => (replacement) => {
     let parts = oldSlug.split("/");
     parts[index] = replacement;
     return parts.join("/");
   };
-    
   const replaceServiceMeshInSlug = replaceSlugPart(3)(chapterData.fields.slug);
+  const replaceChapterInSlug = (slugWithReplacedMesh) => replaceSlugPart(4)(slugWithReplacedMesh);
 
 
   const isMeshActive = (sm) => chapterData.fields.slug.split("/")[3] === sm;
 
+  const mapMeshWithFormattedSlug = (sm, serviceMeshes) => {
+    let chapterFound = false;
+    tableOfContents.forEach(toc => {
+      if(toc.section === sm.fields.section){
+        if(toc.chapter === chapterData.fields.slug.split("/")[4]) chapterFound = true;
```

Comment:
let's move the body of if condition to the next line, just for ease of reading. Or is it only me, who doesn't find that comfortable? 😅

Reply:
oops 😅